votor/src/voting_utils.rs

```diff
@@ -233,8 +249,14 @@ pub(crate) fn insert_vote_and_create_bls_message(
     let bank = context.root_bank_cache.root_bank();
     let bls_message = match generate_vote_tx(&vote, &bank, context) {
         GenerateVoteTxResult::BLSMessage(bls_message) => bls_message,
+        e @ GenerateVoteTxResult::NoAuthorizedVoter(_, _)
+        | e @ GenerateVoteTxResult::NoVoteState(_)
+        | e @ GenerateVoteTxResult::VoteAccountNotFound(_)
+        | e @ GenerateVoteTxResult::Tx(_) => {
+            return Err(VoteError::InvalidConfig(Box::new(e)));
+        }
```

Comment:
For maintainability/avoiding shooting ourselves in the foot, it would be nice to force anybody who adds a new `GenerateVoteTxResult` to bucket it into the right category

Maybe add a `is_invalid_config()`  and `is_transient_error` functions on `GenerateVoteTxResult` that goes through every case?

Reply:
Done.